docs/kap_proposals/kap_11_hashicorp_vault_transit.md

```diff
@@ -0,0 +1,151 @@
+* VAULT_TOKEN: token for vault or file (~/.vault-tokens)
+* VAULT_ROLE_ID: required by approle
+* VAULT_SECRET_ID: required by approle
+* VAULT_USERNAME: username to login to vault
+* VAULT_PASSWORD: password to login to vault
+* VAULT_CLIENT_KEY: the path to an unencrypted PEM-encoded private key matching the client certificate
+* VAULT_CLIENT_CERT: the path to a PEM-encoded client certificate for TLS authentication to the Vault server
+* VAULT_CACERT: the path to a PEM-encoded CA cert file to use to verify the Vault server TLS certificate
+* VAULT_CAPATH: the path to a directory of PEM-encoded CA cert files to verify the Vault server TLS certificate
+* VAULT_NAMESPACE: specify the Vault Namespace, if you have one
+
+Considering any stringdata like `any.value:whatever-you_may*like` ( in our case let’s encrypt `any.value:whatever-you_may*like` with vault transit ) using the key `2022-02-13-test` in a transit secret engine with mount `mytransit` on the vault server, to use this as a secret either follow:
+
+```shell
+$ echo "any.value:whatever-you_may*like" > somefile.txt
+$  kapitan refs --write vaulttransit:<target_name>/to/secret_inside_kapitan --file somefile.txt --target <target_name>
+```
+or in a single line
+```shell
+$ echo "any.value:whatever-you_may*like"  | kapitan refs --write vaulttransit:<target_name>/to/secret_inside_kapitan -t <target_name> -f -
+```
+The entire string __"any.value:whatever-you_may*like"__ will be encrypted by vault and looks like this in return: `vault:v2:Jhn3UzthKcJ2s+sEiO60EUiDmuzqUC4mMBWp2Vjg/DGl+GDFEDIPmAQpc5BdIefkplb6yrJZq63xQ9s=`. This then gets base64 encoded and stored in the secret_inside_kapitan. Now secret_inside_kapitan contains the following
+
+```yaml
+data: dmF1bHQ6djI6SmhuM1V6dGhLY0oycytzRWlPNjBFVWlEbXV6cVVDNG1NQldwMlZqZy9ER2wrR0RGRURJUG1BUXBjNUJkSWVma3BsYjZ5ckpacTYzeFE5cz0=
+encoding: original
+type: vaulttransit
+vault_params:
+  VAULT_ADDR: http://127.0.0.1:8200
+  VAULT_SKIP_VERIFY: 'True'
+  VAULT_TOKEN: s.i53a1DL83REM61UxlJKLdQDY
+  auth: token
+  crypto_key: key
+  mount: transit
+  always_latest: false
+```
+
+Encoding tells the type of data given to kapitan, if it is `original` then after decoding base64 we'll get the original secret and if it is `base64` then after decoding once we still have a base64 encoded secret and have to decode again.
+Parameters in the secret file are collected from the inventory of the target we gave from CLI `--target my_target`. If target isn't provided then kapitan will identify the variables from the environment, but providing `auth` is necessary as a key inside target parameters like the one shown:
+```yaml
+parameters:
+  kapitan:
+    vars:
+      target: my_target
+      namespace: my_namespace
+    secrets:
+      vaulttransit:
+        VAULT_ADDR: http://vault.example.com:8200
+        VAULT_TOKEN: s.i53a1DL83REM61UxlJKLdQDY
```

Comment:
Same for the inventory

Reply:
see also: https://github.com/kapicorp/kapitan/pull/826#discussion_r829709083